src/main/java/carpetextra/mixins/RedstoneWireBlockMixin_pistonRedirectsRedstone.java

```diff
@@ -0,0 +1,29 @@
+package carpetextra.mixins;
+
+import carpetextra.CarpetExtraSettings;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.PistonBlock;
+import net.minecraft.block.RedstoneWireBlock;
+import net.minecraft.util.math.Direction;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(RedstoneWireBlock.class)
+public abstract class RedstoneWireBlockMixin_pistonRedirectsRedstone
+{
+    @Inject(
+            method = "Lnet/minecraft/block/RedstoneWireBlock;connectsTo(Lnet/minecraft/block/BlockState;Lnet/minecraft/util/math/Direction;)Z",
+            at = @At(value = "HEAD"),
+            cancellable = true
+    )
+    private static void onConnectsTo(BlockState state, Direction dir, CallbackInfoReturnable<Boolean> cir)
+    {
+        if (state.isOf(Blocks.PISTON) && CarpetExtraSettings.pistonRedirectsRedstone) {
```

Comment:
please start with a boolean rule check - to prioritize performance with rule turned off. Not big of a difference, but still.
```suggestion
        if ( CarpetExtraSettings.pistonRedirectsRedstone && state.isOf(Blocks.PISTON)) {
```